### PR TITLE
clipit: fix build on gcc14

### PR DESCRIPTION
**Pull request by SFrijters**
## Things done

Package is not very alive, so this is probably the easiest way to unbreak it.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/clipit/package.nix [hunk 1]
```diff
@@ -27,16 +27,21 @@ stdenv.mkDerivation rec {
     intltoolize --copy --force --automake
   '';
 
+  # Fix build on GCC 14
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isGNU "-Wno-error=implicit-function-declaration -Wno-error=incompatible-pointer-types";
```
Comment (paparodeo) [inline pkgs/by-name/cl/clipit/package.nix:31]: looks like Debian has patches to fix these:
https://salsa.debian.org/debian/clipit/-/tree/main/debian/patches?ref_type=heads

Comment (paparodeo) [reply pkgs/by-name/cl/clipit/package.nix:31]: <details><summary>gcc-14-fixes.diff</summary>

```diff
diff --git a/pkgs/by-name/cl/clipit/package.nix b/pkgs/by-name/cl/clipit/package.nix
index c19ae440579d..0ad4f7fe256d 100644
--- a/pkgs/by-name/cl/clipit/package.nix
+++ b/pkgs/by-name/cl/clipit/package.nix
@@ -1,5 +1,6 @@
 {
   fetchFromGitHub,
+  fetchpatch,
   lib,
   stdenv,
   autoreconfHook,
@@ -23,6 +24,19 @@ stdenv.mkDerivation rec {
     sha256 = "0byqz9hanwmdc7i55xszdby2iqrk93lws7hmjda2kv17g34apwl7";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "gcc-14-fixes.patch";
+      url = "https://salsa.debian.org/debian/clipit/-/raw/f2e281ba0247ed8cfc1f6bae916dfac0a9c57c99/debian/patches/missing-prototypes.patch";
+      hash = "sha256-UD183IjV5BprPHQK9bhmUBKfUYgqEZ9M1cRE+AmhAPA=";
+    })
+    (fetchpatch {
+      name = "gcc-14-fixes.patch";
+      url = "https://salsa.debian.org/debian/clipit/-/raw/f2e281ba0247ed8cfc1f6bae916dfac0a9c57c99/debian/patches/incompatible-pointer-types.patch";
+      hash = "sha256-STI1fpnoPdEqu1embQcUlTG712HPbJ+LPm930P13Ixo=";
+    })
+  ];
+
   preConfigure = ''
     intltoolize --copy --force --automake
   '';

```
</details>

**Review by paparodeo — APPROVED** [review]
LGTM. tested build on x64 linux

**Comment by SFrijters** [comment]
Thanks for the suggestion, I like this a lot better as well! I was actually looking for Debian's build scripts / patches, but apparently I was looking in the wrong place. Bookmarking the correct place now :)

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 369068` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clipit</li>
  </ul>
</details>

**Pull request merged**